sdk/python/touca/cli/server.py

```diff
@@ -45,98 +71,71 @@ def find_install_dir():
     compose_file = find_compose_file(install_dir)
     logger.info("Updating docker images")
-    if run_compose(compose_file, "pull"):
+    if compose.run(compose_file, "pull"):
         raise RuntimeError("failed to update docker images")
     logger.info("Starting new containers")
-    if run_compose(compose_file, "up", "-d", "--remove-orphans"):
+    if compose.run(compose_file, "up", "-d", "--remove-orphans"):
         raise RuntimeError("failed to start new containers")
 
 
 def ask(question: str, default: str = None):
     from colorama import Style
 
     msg = f' {Style.DIM}(default is "{default}"){Style.RESET_ALL}' if default else ""
-    output = input(f"$ {question}{msg}\n> ")
+    try:
+        output = input(f"$ {question}{msg}\n> ")
+    except KeyboardInterrupt:
+        raise RuntimeError()
     return output if output else default if default else ""
 
 
-def run_external_command(cmd: List[str], envvars: dict = {}):
-    from os import environ
-    from subprocess import Popen
-    from sys import stderr, stdout
-
-    env = environ.copy()
-    env.update(envvars)
-    proc = Popen(cmd, env=env, universal_newlines=True, stdout=stdout, stderr=stderr)
-    return proc.wait()
-
-
-def run_compose(compose_file: Path, *argv):
-    from os import getuid
-    from pwd import getpwuid
-
-    user = getpwuid(getuid())
-    return run_external_command(
-        [
-            "docker-compose",
-            "-f",
-            compose_file,
-            "-p",
-            "touca",
-            "--project-directory",
-            compose_file.parent.parent,
-            *argv,
-        ],
-        {"UID_GID": f"{user.pw_uid}:{user.pw_gid}"},
-    )
-
-
 def install_file(install_dir: Path, filepath: str):
-    import requests
+    from urllib.request import urlopen
 
     dst_file = install_dir.joinpath(filepath)
     dst_file.parent.mkdir(parents=True, exist_ok=True)
     remote = "https://raw.githubusercontent.com/trytouca/trytouca/main"
-    response = requests.get(f"{remote}/{filepath}")
-    dst_file.write_text(response.text)
+    with urlopen(f"{remote}/{filepath}") as response:
+        dst_file.write_bytes(response.read())
 
 
 def check_server_status(*, attempts=1, port=8080):
+    from urllib.request import urlopen
+    from json import loads
     from time import sleep
 
-    import requests
-
     for attempt in range(1, attempts + 1):
         try:
-            response = requests.get(f"http://localhost:{port}/api/platform", timeout=1)
-            if response.status_code == 200 and response.json()["ready"]:
-                logger.info("Touca server is up and running")
-                return True
+            with urlopen(
+                f"http://localhost:{port}/api/platform", timeout=1
+            ) as response:
+                if response.getcode() == 200 and loads(response.read())["ready"]:
+                    logger.info("Touca server is up and running")
+                    return True
```

Comment:
Unrelated change: switching from `requests` to `urlopen` since this is a tiny operation. I think we now only use `requests` in `touca run` which we could consider removing at some point.